### PR TITLE
Add overwrite and with_tree option for download

### PR DESCRIPTION
**Pull request by appukuttan-shailesh**
Adds option to overwrite existing files, if any, during file download.

### CODE REVIEW

**Comment by appukuttan-shailesh** [comment]
Yes, you are right. (It seemed to have fixed my original problem at the time... but probably it was some other code fix).

Do you suggest we raise an exception, or simply display an error message, such as:
`Destination file already exists! Set 'overwrite=True' to overwrite file.`

Maybe the exception with the traceback/technical terms look a bit scary to the user?

**Comment by appukuttan-shailesh** [comment]
... and return `None`

**Comment by appukuttan-shailesh** [comment]
Also, is there a need for:
`local_directory = os.path.join(os.path.abspath(local_directory), *os.path.dirname(file_path).split("/"))`
Wouldn't it be better to have the local directory as the current directory (default), or that specified by the user? Currently if I wish to download a specific file, the entire subtree gets created at the target location.

**Comment by appukuttan-shailesh** [comment]
Updated: throws exception and also provides an option to specify if download with subtree or not.

File: hbp_archive.py [hunk 1]
```diff
@@ -187,15 +187,18 @@ def size(self, units='bytes'):
         """Total size of all data in the container"""
         return scale_bytes(int(self.metadata['x-container-bytes-used']), units)
 
-    def download(self, file_path, local_directory="."):
+    def download(self, file_path, local_directory=".", with_tree=True, overwrite=False):
         """Download a file from the container"""
```
Comment (apdavison) [inline hbp_archive.py]: Please could you document the new options in the docstring?

**Pull request merged**